### PR TITLE
Add mixed decimal testing for binary arithmetic ops

### PR DESCRIPTION
**Pull request by rwlee**
Followup to #7454
`pmod` tests are xfailed due to #7553 -- was not able to find a simple cause or solution to the #7553 problem so did not include any form of a solution to it in this PR.

### CODE REVIEW

File: integration_tests/src/main/python/arithmetic_ops_test.py [hunk 1]
```diff
@@ -375,6 +377,29 @@ def test_mod_pmod_by_zero_not_ansi(data_gen):
             'cast(-12 as {}) % cast(0 as {})'.format(string_type, string_type)),
         {'spark.sql.ansi.enabled': 'false'})
 
+@pytest.mark.parametrize('lhs', [byte_gen, short_gen, int_gen, long_gen, DecimalGen(6, 5),
+    DecimalGen(6, 4), DecimalGen(5, 4), DecimalGen(5, 3), DecimalGen(4, 2), DecimalGen(3, -2),
+    DecimalGen(16, 7), DecimalGen(19, 0), DecimalGen(30, 10)], ids=idfn)
+@pytest.mark.parametrize('rhs', [byte_gen, short_gen, int_gen, long_gen, DecimalGen(6, 3),
+    DecimalGen(10, -2), DecimalGen(15, 3), DecimalGen(30, 12), DecimalGen(3, -3),
+    DecimalGen(27, 7), DecimalGen(20, -3)], ids=idfn)
+def test_mod_mixed_decimal(lhs, rhs):
```
Comment (razajafri) [inline integration_tests/src/main/python/arithmetic_ops_test.py:386]: name should reflect that it's not just for decimals

File: integration_tests/src/main/python/arithmetic_ops_test.py [hunk 2]
```diff
@@ -375,6 +377,29 @@ def test_mod_pmod_by_zero_not_ansi(data_gen):
...
+# @pytest.mark.xfail(reason='BUG https://github.com/NVIDIA/spark-rapids/issues/7553')
+@pytest.mark.parametrize('lhs', [byte_gen, short_gen, int_gen, long_gen, DecimalGen(6, 5),
+    DecimalGen(6, 4), DecimalGen(5, 4), DecimalGen(5, 3), DecimalGen(4, 2), DecimalGen(3, -2), 
+    DecimalGen(16, 7), DecimalGen(19, 0), DecimalGen(30, 10)
+    ], ids=idfn)
+@pytest.mark.parametrize('rhs', [ byte_gen, short_gen, int_gen, long_gen, DecimalGen(6, 3),
+    DecimalGen(10, -2), DecimalGen(15, 3), DecimalGen(30, 12), DecimalGen(3, -3),
+    DecimalGen(27, 7), DecimalGen(20, -3)
+    ], ids=idfn)
+def test_pmod_mixed_decimal(lhs, rhs):
```
Comment (razajafri) [inline integration_tests/src/main/python/arithmetic_ops_test.py:399]: same comment as above

**Review by revans2 — DISMISSED** [review]
I agree with @razajafri but for me it is a nit.

File: integration_tests/src/main/python/arithmetic_ops_test.py [hunk 3]
```diff
@@ -113,11 +113,12 @@ def test_subtraction(data_gen):
...
-@pytest.mark.parametrize('rhs', [DecimalGen(6, 3), DecimalGen(10, -2), DecimalGen(15, 3),
-    DecimalGen(30, 12), DecimalGen(3, -3), DecimalGen(27, 7), DecimalGen(20, -3)], ids=idfn)
+@pytest.mark.parametrize('lhs', [byte_gen, short_gen, int_gen, long_gen, DecimalGen(6, 5),
+    DecimalGen(6, 4), DecimalGen(5, 4), DecimalGen(5, 3), DecimalGen(4, 2), DecimalGen(3, -2),
+    DecimalGen(16, 7), DecimalGen(19, 0), DecimalGen(30, 10)], ids=idfn)
+@pytest.mark.parametrize('rhs', [byte_gen, short_gen, int_gen, long_gen, DecimalGen(6, 3),
+    DecimalGen(10, -2), DecimalGen(15, 3), DecimalGen(30, 12), DecimalGen(3, -3),
+    DecimalGen(27, 7), DecimalGen(20, -3)], ids=idfn)
 @pytest.mark.parametrize('addOrSub', ['+', '-'])
 def test_addition_subtraction_mixed_decimal(lhs, rhs, addOrSub):
```
Comment (razajafri) [inline integration_tests/src/main/python/arithmetic_ops_test.py:123]: Here too

**Comment by rwlee** [comment]
build

**Comment by rwlee** [comment]
build

**Review by revans2 — APPROVED** [review]

**Review by gerashegalov — APPROVED** [review]
LGTM

**Pull request merged**